cdap-app-templates/cdap-etl/cdap-etl-api/src/main/java/io/cdap/cdap/etl/api/Joiner.java

```diff
@@ -36,8 +39,21 @@
    * @return returns join key
    * @throws Exception if there is some error getting the join key
    */
+  @Deprecated
   JOIN_KEY joinOn(String stageName, INPUT_RECORD inputRecord) throws Exception;
```

Comment:
can make this default as well, where the default implementation throws an exception. That way, plugins are not forced to implement this deprecated method.

Reply:
done